presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -403,7 +403,12 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                 leftSource = new ProjectNode(idAllocator.getNextId(), leftSource, leftProjections.build());
                 rightSource = new ProjectNode(idAllocator.getNextId(), rightSource, rightProjections.build());
 
-                output = new JoinNode(node.getId(), node.getType(), leftSource, rightSource, joinConditionBuilder.build(), newJoinFilter, node.getLeftHashSymbol(), node.getRightHashSymbol());
+                List<Symbol> outputSymbols = ImmutableList.<Symbol>builder()
```

Comment:
Can new join be cross join? If so, it should have all symbols from left and right sources. Therefore I think `PruneUnreferencedOutputs` should do pruning (it checks if join is cross join)